Added more image format handling

n3o2k7i8ch5 opened the pull request:
Added asset images and svg images rendering.

alihassan143 commented:
@n3o2k7i8ch5 
kindly be specific what did you fixed or added in the code 
don't add same code with just spaces

alihassan143 commented:
closing this pr

n3o2k7i8ch5 commented:
Fixed loads of small issues and simplified the code, reusing some functions in place of repetitions.

Needed to use this library for generating pdfs with a wide variety of html objects, often nested.
